backend/books/views.py

```diff
@@ -88,6 +95,16 @@ def getOrCreateModel(self, item_list, model):
             obj, created = model.objects.get_or_create(
                 name=item.strip(),
             )
+    
+    def get_queryset(self):
+        default_query_set = Book.objects.all()
+        default_query_set = default_query_set.annotate(author=Max('authors__name'))
+        default_query_set = default_query_set.annotate(genre=Max('genres__name'))
+
+        if genre := self.request.query_params.get('genre'):
+            return default_query_set.filter(genres__name=genre)
```

Comment:
Would you mind just explaining what these line are doing? Did some research, but a bit confused, especially about the `Max` function being used. And what is the if statement for/why does it only apply to genres?

Reply:
just added a comment to annotations and genre filtering